pandas/core/indexes/interval.py

```diff
@@ -514,6 +519,41 @@ def _maybe_cast_indexed(self, key):
 
         return key
 
+    def _maybe_convert_i8(self, key):
+        if isinstance(key, Interval):
```

Comment:
can you document this a bit, and can you simplify at all?

Reply:
Done. Look better?